subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/rest/HttpServer.kt

```diff
@@ -5,66 +5,81 @@ import java.io.BufferedReader
 import java.io.InputStreamReader
 import java.io.PrintWriter
 import java.net.ServerSocket
+import java.net.Socket
 
-internal class HttpServer(private val handlers: List<RequestHandler>, private val debug: Boolean) {
+internal class HttpServer(
+    private val handlers: List<RequestHandler>,
+    private val debug: Boolean
+) {
 
-    private var serverThread: Thread? = null
+    private var serverSocket: ServerSocket? = null
 
-    @Volatile
-    private var isStopped = false
```

Comment:
I expected `isStopped` to be used in while loop instead of `!Thread.interrupted()`, but anyways it wasn't working as expected, because of blocking nature of `ServerSocket.accept()`